### PR TITLE
Add concept definitions

### PR DESCRIPTION
**Pull request by TheOneAndra**
Fix #Issue

**Changes proposed in this pull request**:
- Define Action
- Define Requirement
- Adapt Attribute
- Harmonize concept def

The following steps were realized, as well (if applies):
- [ ] Update the CHANGELOG.md
- [ ] For new functionalities: explain in readthedocs

<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl
-institut/open_plan/blob/dev/CONTRIBUTING.md).*<sub>

### CODE REVIEW

File: docs/tool_interface.rst [hunk 1]
```diff
@@ -53,10 +65,6 @@ widget
     Smaller window within the tool's :ref:`main-window-def`.
     A *widget* can be moved around by the user within the :ref:`main-window-def` and collapsed into another *widget* (then each *widget* is accessible via tabs).
```
Comment (TheOneAndra) [inline docs/tool_interface.rst:70]:
```suggestion

.. _views-label:

```

**Review by TheOneAndra — COMMENTED** [review]
Deleted by mistake, I add it again here with a suggestion

File: docs/tool_interface.rst [hunk 2]
```diff
@@ -11,50 +11,60 @@ In order to describe the interface, a few definitions are required at first.
...
+
+Action
+------
+    An action describes what the user can do in a certain :ref:`view-def` or :ref:`view-component-def`. Each action has an index which is used to link it to an :ref:`attribute-def`.
+
+.. _requirement-def:
+
+_Requirement
+------------
+    A requirement describes necessary features of a :ref:`view-def`, a :ref:`view-component-def` or a :ref:`action-def`. A requirement can specify required actions by the user as well as describing what happens if a requirement is not met. Each action has an index which is used to link it to an :ref:`attribute-def`.
```
Comment (Bachibouzouk) [inline docs/tool_interface.rst]:
```suggestion
    A requirement describes specific conditions of a :ref:`view-def`, a :ref:`view-component-def`, an :ref:`action-def` or an :ref:`attribute-def` which need to be met. Each requirement has an index which is used to link it to an :ref:`attribute-def`.
```

Comment (Bachibouzouk) [reply docs/tool_interface.rst]: I would provide an example of a simple requirement and a simple action to help the readers

Comment (Bachibouzouk) [reply docs/tool_interface.rst]: Examples

Action: the user can click on button A to trigger action B

Requirement: the button B cannot be clicked unless text input C is not filled by user

File: docs/tool_interface.rst [hunk 3]
```diff
@@ -11,50 +11,60 @@ In order to describe the interface, a few definitions are required at first.
 
 .. _view-def:
 
-view
+View
```
Comment (Bachibouzouk) [inline docs/tool_interface.rst]: when we refer to them in the text, we'll get capital letters then, I don't know what I prefer ...

Comment (TheOneAndra) [reply docs/tool_interface.rst]: Agree, it should not be capital letters when refered in text. It is weird to have non-caps h3 while h1 and h2 are caps but so it is.

File: docs/tool_interface.rst [hunk 4]
```diff
@@ -13,37 +13,51 @@ In order to describe the interface, a few definitions are required at first.
...
+action
+------
+    An action describes what the user can do in a certain :ref:`view-def` or :ref:`view-component-def`. Each action has an index which is used to link it to an :ref:`attribute-def`.
+Example: [index] The user can click on button B to trigger action C
+
+.. _requirement-def:
+
+requirement
+------------
+    A requirement describes specific conditions of a :ref:`view-def`, a :ref:`view-component-def`, an :ref:`action-def` or an :ref:`attribute-def`. A requirement specifies necessary actions that need to be done by the user and explains what happens if the requirement is not met. Each requirement has an index which is used to link it to an :ref:`attribute-def`.
```
Comment (Bachibouzouk) [inline docs/tool_interface.rst:42]: I am not sure what is the meaning/use of this sentence : "A requirement specifies necessary actions that need to be done by the user and explains what happens if the requirement is not met."

Comment (Bachibouzouk) [inline docs/tool_interface.rst:42]:
```suggestion
.. TODO : rework this sentence
    A requirement describes specific conditions of a :ref:`view-def`, a :ref:`view-component-def`, an :ref:`action-def` or an :ref:`attribute-def`. A requirement specifies necessary actions that need to be done by the user and explains what happens if the requirement is not met. Each requirement has an index which is used to link it to an :ref:`attribute-def`.
```

**Review by Bachibouzouk — APPROVED** [review]
I merge it for now, @TheOneAndra , can you rework the sentence where I inserted the TODO in a subsequent PR?

**Pull request merged**